### PR TITLE
Generic and chain agnostic upgrade script for all contracts

### PR DESCRIPTION
**Pull request by blockgroot**
Command to upgrade all the contracts mentioned in address.json at once : `npx hardhat run scripts/safe-scripts/upgradeAll.ts --network <network_name>`

### CODE REVIEW

File: contracts/L2/ETHx.sol
Comment (jac18281828) [inline contracts/L2/ETHx.sol:1]: Can you revert this change here.   I created an issue https://github.com/stader-labs/ethx/issues/229

This is to remove the L2 implementation from ETHx.    It would be better to have clean chain of custody on this change in case we wish to put it back.

**Review by jac18281828 — COMMENTED** [review]
This looks ready to approve to me.  Just revert the L2/ETHx change and we can merge this

File: contracts/L2/ETHx.sol
Comment (blockgroot) [reply contracts/L2/ETHx.sol:1]: [a4d4564](https://github.com/stader-labs/ethx/pull/237/commits/a4d456405b088fe4294908616a4e723e15028035)

**Review by jac18281828 — APPROVED** [review]
Thank you!  These release process changes are essential.  😇

**Review by jac18281828 — APPROVED** [review]
reapprove after rebasing
